### PR TITLE
Subscriptions: Don't reload page if email field is empty

### LINKED ISSUE #1793: Subscriptions: Clicking on Subscribe button without having entered an email address reloads the page
If you have the Subscription widget and click the Subscribe button without having added an email address to the field, it simply reloads the page. Ideally, it'd avoid the page reload and display a message notifying the user that they should add an email address.

### PR DESCRIPTION
**Pull request by dereksmart**
Fixes #1793

Added 2 checks - one HTML5 required attr, and some custom js (Safari doesn't respect the HTML `required` attr)

### CODE REVIEW

**Comment by dereksmart** [comment]
@georgestephanis can you please take a look at this?  I don't see how this PR could cause those errors.

File: modules/subscriptions.php [hunk 1]
```diff
@@ -645,7 +645,7 @@ function widget( $args, $instance ) {
 						<label id="jetpack-subscribe-label" for="<?php echo esc_attr( $subscribe_field_id ); ?>">
 							<?php echo !empty( $subscribe_placeholder ) ? esc_html( $subscribe_placeholder ) : esc_html__( 'Email Address:', 'jetpack' ); ?>
 						</label>
-						<input type="email" name="email" value="<?php echo esc_attr( $subscribe_email ); ?>" id="<?php echo esc_attr( $subscribe_field_id ); ?>" placeholder="<?php echo esc_attr( $subscribe_placeholder ); ?>" />
+						<input type="email" name="email" required class="required" value="<?php echo esc_attr( $subscribe_email ); ?>" id="<?php echo esc_attr( $subscribe_field_id ); ?>" placeholder="<?php echo esc_attr( $subscribe_placeholder ); ?>" />
```
Comment (samhotchkiss) [inline modules/subscriptions.php]: Proper html5 would dictate: `required="required"`, right?  Values for everything!

Comment (dereksmart) [inline modules/subscriptions.php]: Ah yep, it's XHTML that says we need values for everything.  Pushing a change now.  

https://make.wordpress.org/core/handbook/coding-standards/html/#quotes

Comment (samhotchkiss) [inline modules/subscriptions.php]: One of those HTMLs ;-)

**Comment by samhotchkiss** [comment]
Looks good to me

**Pull request merged**